### ISSUE TITLE
Icons are hard to see in selected states

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#55857]
The new list/tree selection color in light theme does not work well for me. The dark blue color looks off compared to the (probably unchanged) focus color. And worse, I feel I can no longer see the twistie in this case:

![image](https://user-images.githubusercontent.com/900690/43705949-4894613e-9964-11e8-99ec-da92648fa1c1.png)

**Comment by bpasero** [comment]
@misolori I think we should change the color of the twistie to white though, that is the same macOS seems to do:

![image](https://user-images.githubusercontent.com/900690/43776504-65107946-9a50-11e8-8f17-c3095e465495.png)

**Comment by misolori** [comment]
@bpasero yes, would love to do this! I'll need to investigate how to do this, here's a quick mockup:

![image](https://user-images.githubusercontent.com/35271042/43790857-b2ec1670-9a28-11e8-8289-89621fb067a0.png)

**Comment by bpasero** [comment]
@misolori looks good

**Comment by bpasero** [comment]
@misolori also looks like we need to take care of this guy:

![image](https://user-images.githubusercontent.com/900690/44139122-6e8d56a2-a076-11e8-99dc-9d88813e0c28.png)

**Comment by bpasero** [comment]
Also an issue for our octicons for extensions it seems:

![image](https://user-images.githubusercontent.com/900690/44202052-37778900-a14b-11e8-853b-59f3161419a1.png)

**Comment by bpasero** [comment]
Just also noticed our Git actions in dark having a bad contrast now:

![image](https://user-images.githubusercontent.com/900690/44267841-fefdab00-a22f-11e8-94ba-39235bc1c09e.png)

**Comment by misolori** [comment]
@bpasero I've got a branch [`misolori/light-theme-icons-selection`](https://github.com/Microsoft/vscode/tree/misolori/light-theme-icons-selection) that covers most of these fixes: 

![image](https://user-images.githubusercontent.com/35271042/44422541-4b026580-a538-11e8-9787-8ae2db68d6cf.png)

![image](https://user-images.githubusercontent.com/35271042/44422520-3e7e0d00-a538-11e8-8e03-979fb13c3626.png)

![image](https://user-images.githubusercontent.com/35271042/44422583-5e153580-a538-11e8-97eb-789b0ace7bbb.png)

![image](https://user-images.githubusercontent.com/35271042/44422886-4ee2b780-a539-11e8-96f3-b0e9d94730b5.png)



The only ones that I'm unable to fix are the extension icons as those are set in `package.json` and there is no way to specify what icon to use when selected.

What are your thoughts about setting extension icons via `-webkit-mask` so that they are able to be themeable? I would imagine they'd inherit the `list.activeSelectionForeground` color.

**Comment by bpasero** [comment]
> We can also think about applying the `list.activeSelectionForeground` on the icons when selected.

I think that is probably the right solution using the same webkit-mask trick we already use to allow to set the foreground color of activity bar icons. 

/cc @joaomoreno

**Comment by misolori** [comment]
@joaomoreno would you be able to help out with this? I think we'd want to make all of the icons in the list widget use the `webkit-mask` trick so they're consistent.

**Comment by joaomoreno** [comment]
The only icons the tree has are: twistie and loading indicator. All other icons belong to the many users of the tree.

**Comment by misolori** [comment]
Made some progress on this, though I've ran into a few critical issues when I tried converting the icons into a `-webkit-mask`:

- This assumes that all list widget icons are non-images, which means contributions with avatars will break (i.e. github pr extension)
- This assumes that all icons are the same single color, which means multi-colored icons will visually break
- Adjusting the list widget actions (like search and scm list actions) touches other areas (editor toolbar) which means it forces those icons and contributions to also inherit the issues above

So I can make a PR that will fix the basic list widget icons that we own (twisties and loading indicator) but am unsure of how to solve the rest of the issues above.

**Comment by isidorn** [comment]
I think this is a bit better with the new icons.
Though maybe the contrast could still be better. So ping @misolori

**Comment by misolori** [comment]
Not much has changed in terms of color but since the icons are thin now they may be improved a bit:

<img width="307" alt="image" src="https://user-images.githubusercontent.com/35271042/59776619-fd45af00-9267-11e9-98b4-a7a64eb5b4dc.png">

We still need to do the heavy work by styling our icons dynamically that I'm hoping will happen after we update our icons.

**Comment by misolori** [comment]
Verification steps:

Ensure that the icons in the tree widget inherit the foreground color changes from their elements like so:

<img width="313" alt="image" src="https://user-images.githubusercontent.com/35271042/65707057-95f68280-e040-11e9-886a-4be1e4a02db5.png">

_Note: this does not include icons from contribution points like extensions + SCM actions_

**Comment by isidorn** [comment]
Awesome! The twistie is lovely

**Comment by bpasero** [comment]
Great 👍
